add_zero_quantity/tests/test_zero_quantity.py

```diff
@@ -0,0 +1,81 @@
+from odoo.tests.common import TransactionCase
+from odoo.tests import tagged
+from odoo import Command
+
+@tagged("post_install", "-at_install")
+class TestIsZeroQuantity(TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.partner = cls.env['res.partner'].create({'name': 'Test Partner'})
+
+        cls.product = cls.env['product.product'].create({
+            'name': 'Test Product',
+            'type': 'consu'
+        })
+
+        cls.move = cls.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': cls.partner.id
+        })
+
+        cls.move_line = cls.env['account.move.line'].create({
+            'move_id': cls.move.id,
+            'product_id': cls.product.id,
+            'quantity': 10,
+            'price_unit': 100,
+            'is_zero_quantity': False
+        })
+
+    def test_get_l10n_in_edi_line_details(self):
+        """Test that the 'Qty' field is set to zero when 'is_zero_quantity' is True."""
+        edi_format = self.env['account.edi.format'].create({
+            'name': 'Test EDI Format',
+            'code': 'TEST_CODE'
+        })
+        line_details = edi_format._get_l10n_in_edi_line_details(0, self.move_line, {})
+        self.assertNotEqual(
+            line_details.get('Qty'),
+            0,
+            "Quantity should not be zero by default"
+        )
+
+        self.move_line.is_zero_quantity = True
+        line_details = edi_format._get_l10n_in_edi_line_details(0, self.move_line, {})
+        self.assertEqual(
+            line_details.get('Qty'),
+            0,
+            "Quantity should be zero when is_zero_quantity is set"
+        )
+
+    def test_get_gstr1_hsn_json(self):
+        """Test that 'qty' in HSN JSON is set to zero when 'is_zero_quantity' is True."""
+        gst_return = self.env['l10n_in.gst.return.period'].create({'name': 'Test GST Return'})
```

Comment:
```suggestion
        gst_return = self.env['l10n_in.gst.return.period'].create()
```
Please give the default params atleast